docs/src/main/asciidoc/javascript/config.js

```diff
@@ -2,192 +2,381 @@ jQuery(function(){
+        var descriptions = table.querySelectorAll(".description");
+        if(descriptions){
+            for (description of descriptions){
+                makeCollapsible(input, description);
             }
-        });
+        }
+    }
+}
+
+function initiateSearch(event){
+    // only start searching after the user stopped typing for 300ms, since we can't abort
+    // running tasks, we don't want to search three times for "foo" (one letter at a time)
+    if(typingTimer)
+        clearTimeout(typingTimer);
+    typingTimer = setTimeout(() => search(event.target), 300)
+}
+
+function highlight(element, text){
+    var iter = document.createNodeIterator(element, NodeFilter.SHOW_TEXT, null);
+
+    while (n = iter.nextNode()){
+        var parent = n.parentNode;
+        var elementText = n.nodeValue;
+        if(elementText == undefined)
+            continue;
+        var elementTextLC = elementText.toLowerCase(); 
+        var index = elementTextLC.indexOf(text);
+        if(index != -1){
+            var start = 0;
+            var fragment = document.createDocumentFragment()
+            // we use the DOM here to avoid &lt; and such being parsed as elements by jQuery when replacing content
+            do{
+                // text before
+                fragment.appendChild(document.createTextNode(elementText.substring(start, index)));
+                // highlighted text
+                start = index + text.length;
+                var hlText = document.createTextNode(elementText.substring(index, start));
+                var hl = document.createElement("span");
+                hl.appendChild(hlText);
+                hl.setAttribute("class", "configuration-highlight");
+                fragment.appendChild(hl);
+            }while((index = elementTextLC.indexOf(text, start)) != -1);
+            // text after
+            n.nodeValue = elementText.substring(start);
+            // replace
+            parent.insertBefore(fragment, n);
+        }
+    }
+    iter.detach();
+}
+
+function clearHighlights(table){
+    for (var span of table.querySelectorAll("span.configuration-highlight")) {
+        var parent = span.parentNode;
+        var prev = span.previousSibling;
+        var next = span.nextSibling;
+        var target;
+        if(prev && prev.nodeType == Node.TEXT_NODE){
+            target = prev;
+        }
+        var text = span.childNodes.item(0).nodeValue;
+        if(next && next.nodeType == Node.TEXT_NODE){
+            text += next.nodeValue;
+            parent.removeChild(next);
+        }
+        if(target){
+            target.nodeValue += text;
+        }else{
+            target = document.createTextNode(text);
+            parent.insertBefore(target, span);
+        }
+        parent.removeChild(span);
+    }
 }
 
-function search(table, rows, val){
-    var search = val.trim().toLowerCase();
-    var lastSearch = table.data("lastSearch");
+function findText(row, search){
```

Comment:
`Expand` and `All` are special cases since we can them in the DOM as either part of config description or as part of the row expansion feature. I think we'll have to tweak  search function to ignore `all` or `expand` (or their fragments) search inputs  when the row is expandable but its description does not contain `expand` or `all` or their fragments.  

Escaping them will do the trick. Likewise for the highlighting feature.